table/column.go

```diff
@@ -371,7 +373,27 @@ func GetColOriginDefaultValue(ctx sessionctx.Context, col *model.ColumnInfo) (ty
 
 // GetColDefaultValue gets default value of the column.
 func GetColDefaultValue(ctx sessionctx.Context, col *model.ColumnInfo) (types.Datum, error) {
-	return getColDefaultValue(ctx, col, col.GetDefaultValue())
+	defaultValue := col.GetDefaultValue()
+	if !col.DefaultIsExpr {
+		return getColDefaultValue(ctx, col, col.GetDefaultValue())
+	}
+	return getColDefaultExprValue(ctx, defaultValue)
+}
+
+func getColDefaultExprValue(ctx sessionctx.Context, defaultValue interface{}) (types.Datum, error) {
+	x, ok := defaultValue.(string)
+	if !ok {
+		// Todo: refine the error.
+		return types.Datum{}, errors.New("can't cast default value as  expression string")
+	}
+	var defaultExpr ast.ExprNode
+	expr := fmt.Sprintf("select %s", x)
+	charset, collation := charset.GetDefaultCharsetAndCollate()
+	stmts, _, err := parser.New().Parse(expr, charset, collation)
```

Comment:
It is enough to pass empty strings to charset and collation.

A question: is it necessary to parse the expression once again?

Reply:
1. nice idea.
2. Actually we can do the expr cache like `colDefaultVals` does, Given the sequence function hasn't been implemented now, it's hard to do the test, so I push it to the next PR.